src/freenas/usr/local/lib/middlewared_truenas/plugins/failover.py

```diff
@@ -1689,12 +1705,17 @@ async def hook_pool_dataset_unlock(middleware, datasets):
 
 
 async def hook_pool_dataset_post_create(middleware, dataset_data):
-    if dataset_data['encrypted'] and str(dataset_data['key_format']).upper() == 'PASSPHRASE':
-        await middleware.call(
-            'failover.update_encryption_keys', {
-                'datasets': [{'name': dataset_data['name'], 'passphrase': dataset_data['encryption_key']}]
-            }
-        )
+    if dataset_data['encrypted']:
+        if str(dataset_data['key_format']).upper() == 'PASSPHRASE':
```

Comment:
Why do we use string typing here?

Reply:
Because it can be `None` as well if the data was not encrypted

Reply:
What other values are possible there?

Reply:
`None`, `PASSPHRASE`, `HEX`

Reply:
Why `.upper()` then?

Reply:
The actual values will be lowercase because `zfs` wants them like that - a more accurate version of my previous comment is `None`, `passphrase`, `hex`. That's why we use `upper` to ensure we do a normalised match. I can do it the other way around as well to use lowercase `PASSPHRASE` if you think that's better ?

Reply:
No action needed, just thought there were some data inconsistencies